### PR TITLE
[polkadot-runtime-parachains] migrate disputes and disputes/slashing to benchmarking to bench v2 syntax

### PR DESCRIPTION
**Pull request by clangenb**
[polkadot-runtime-parachains] migrate disputes and disputes/slashing to benchmarking to bench v2 syntax

Part of:
* #6202

### CODE REVIEW

**Review by re-gius — APPROVED** [review]

**Comment by clangenb** [comment]
@re-gius  review reminder ;)

File: polkadot/runtime/parachains/src/disputes/slashing/benchmarking.rs [hunk 1]
```diff
@@ -136,27 +141,28 @@ fn dispute_proof(
...
+mod benchmarks {
+	use super::*;
 
 	// in this setup we have a single `ForInvalid` dispute
 	// submitted for a past session
-	report_dispute_lost {
-		let n in 4..<<T as super::Config>::BenchmarkingConfig as BenchmarkingConfiguration>::MAX_VALIDATORS;
-
+	#[benchmark]
+	fn report_dispute_lost(n: Linear<4, { max_validators_for::<T>() }>) {
```
Comment (bkchr) [inline polkadot/runtime/parachains/src/disputes/slashing/benchmarking.rs:151]:
```suggestion
	fn report_dispute_lost_unsigned(n: Linear<4, { max_validators_for::<T>() }>) {
```

File: polkadot/runtime/parachains/src/disputes/slashing/benchmarking.rs [hunk 2]
```diff
@@ -136,27 +141,28 @@ fn dispute_proof(
...
+
+		#[block]
+		{
+			let result = Pallet::<T>::report_dispute_lost_unsigned(
+				origin,
+				Box::new(dispute_proof),
+				key_owner_proof,
+			);
+			assert!(result.is_ok());
+		}
```
Comment (bkchr) [inline polkadot/runtime/parachains/src/disputes/slashing/benchmarking.rs:164]:
```suggestion
		#[extrinsic_call]
		_(origin, Box::new(dispute_proof), key_owner_proof);
```

**Review by bkchr — COMMENTED** [review]
And then in the normal pallet code you need to use `report_dispute_lost_unsigned` as name for the weight function as well.

**Comment by clangenb** [comment]
Thanks @bkchr for the hints! Committed your changes and integrated them with the codebase! 👍

**Comment by clangenb** [comment]
@re-gius This PR has got the other review now. May I ask you to go ahead with the merge?

**Pull request merged**